pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -136,29 +136,30 @@ func (c *Cluster) Start() error {
 				return fmt.Errorf("failed to update mgr deployment %s. %+v", resourceName, err)
 			}
 		}
-	}
 
-	if err := c.configureOrchestratorModules(); err != nil {
-		logger.Errorf("failed to enable orchestrator modules. %+v", err)
-	}
+		if err := c.configureOrchestratorModules(); err != nil {
+			logger.Errorf("failed to enable orchestrator modules. %+v", err)
+		}
 
-	if err := c.enablePrometheusModule(c.Namespace); err != nil {
-		logger.Errorf("failed to enable mgr prometheus module. %+v", err)
-	}
+		if err := c.enablePrometheusModule(c.Namespace); err != nil {
+			logger.Errorf("failed to enable mgr prometheus module. %+v", err)
+		}
 
-	if err := c.configureDashboard(c.dashboardPort()); err != nil {
-		logger.Errorf("failed to enable mgr dashboard. %+v", err)
-	}
+		if err := c.configureDashboard(mgrConfig); err != nil {
+			logger.Errorf("failed to enable mgr dashboard. %+v", err)
+		}
 
-	// create the metrics service
-	service := c.makeMetricsService(appName)
-	if _, err := c.context.Clientset.CoreV1().Services(c.Namespace).Create(service); err != nil {
-		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create mgr service. %+v", err)
+		// create the metrics service
+		service := c.makeMetricsService(appName)
```

Comment:
The call to `makeMetricsService()` can be outside the `for` loop. Inside the loop it would just be ignored after the first time it is called since it would already exist.